src/rust/ide/view/src/documentation.rs

```diff
@@ -148,7 +155,32 @@ impl ViewModel {
     /// Create a container for generated content and embed it with stylesheet.
     fn push_to_dom(&self, content:String) {
         let data_str = format!(r#"<div class="docVis">{}{}</div>"#,documentation_style(),content);
-        self.dom.dom().set_inner_html(&data_str)
+        self.dom.dom().set_inner_html(&data_str);
+    }
+
+    /// Append listeners to copy buttons in doc to enable copying examples.
+    fn append_listeners_to_copy_buttons(&self) {
+        let code_blocks  = self.dom.dom().get_elements_by_class_name("CodeBlock");
+        let copy_buttons = self.dom.dom().get_elements_by_class_name("copyCodeBtn");
+        for i in 0..copy_buttons.length() {
+            let copy_button = copy_buttons.get_with_index(i);
+            let code_block  = code_blocks.get_with_index(i);
+            copy_button.map(|btn| { btn.dyn_into::<HtmlElement>().map(|cpy_btn| {
+                code_block.map(|code| { code.dyn_into::<HtmlElement>().map(|code_blk| {
+                    let closure  = move |_event: MouseEvent| {
+                        let inner_code = code_blk.inner_text();
+                        clipboard::write_text(inner_code);
+                    };
+                    let closure  = Closure::wrap(Box::new(closure).into_fn_mut());
+                    let callback = closure.as_ref().unchecked_ref();
+                    match cpy_btn.add_event_listener_with_callback("click", callback){
+                        Ok(_)    => (),
+                        Err(err) => error!(&self.logger, "Unable to add event listener: {err:?}")
+                    }
+                    closure.forget();
+                }) });
+            }) });
+        }
```

Comment:
1. The ViewModel structure should have a field keeping a vector of Closures. Then in this function we should gather all the closures into this instead of forgetting them, and replace old vector in structure with a new one. Therefore I would also rename this function to `set_listeners_to_copy_buttons`.
2. The loop could be much simplified and improved:
  * Instead of plain loop, lets map each element to option of closure `(0..copy_buttons.length()).iter().map(|i| ...`
  * Then, instead of nested option::map calls you could make use of `?`: 
    ```rust
    let copy_button = copy_buttons.get_with_index(i)?.dyn_into::<HtmlElement>()?;
    let code_block  = code_blocks.get_with_index(i)?.dyn_into::<HtmlElement>()?;
    ...
    ```
* After all we could call filter_map and log all Nones - as having None is not an usual situation, and I want to have an 
  error log for that.